Домашнее задание спринт 19

Nataliya2288 opened the pull request.

Nataliya2288 committed:
Домашнее задание спринт 19

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/root/ui/RootActivity.kt:
```diff
@@ -0,0 +1,36 @@
+
+    override fun toggleBottomNavigationViewVisibility(visible: Boolean) {
+        if (visible) {
+            bottomNavigationView.visibility = View.VISIBLE
```
![view](https://github.com/user-attachments/assets/6ce55c53-3877-485a-9dd5-ab0036a7d1b9)

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -157,24 +184,39 @@ class SearchActivity : AppCompatActivity() {
             }
             false
         }
+        KeyboardVisibilityEvent.setEventListener(
```
Можно написать так
```kotlin
        KeyboardVisibilityEvent.setEventListener(
            activity = requireActivity(),
            lifecycleOwner = viewLifecycleOwner
        ) { isOpen ->
            if (isOpen) {
                onKeyboardVisibilityChanged(true)
            } else {
                onKeyboardVisibilityChanged(false)
            }
        }
```

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -64,51 +72,73 @@ class SearchActivity : AppCompatActivity() {
+        if (context is BottomNavigationListener) {
+            bottomNavigationListener = context
+        } else {
+            throw IllegalArgumentException("Activity must implement BottomNavigationListener")
```
Выглядит так, что  лучше сломать приложение, чем скрыть панель навигации.
Для пет проекта такое нормально, но если это было коммерческое приложение, то падение приложения у пользователя может обернутся куда более серьезными последствиями, чем нескрытая панель.

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -124,9 +154,6 @@ class SearchActivity : AppCompatActivity() {
-        backArrowImageView.setOnClickListener {
-            finish()
-        }
         val textWatcher = object : TextWatcher {
```
![TextWatcher](https://github.com/user-attachments/assets/9530d82d-055d-4263-af26-1cac6f9c6aca)

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -17,18 +19,25 @@ import android.widget.ImageView
+    private var bottomNavigationListener: BottomNavigationListener? = null
+
+    private lateinit var binding: FragmentSearchBinding
     var textFromSearchWidget = ""
```
textFromSearchWidget сделать private

Nataliya2288 committed:
Домашнее задание спринт 19 v2

Nataliya2288 committed:
Домашнее задание спринт 19 v2.1

Nataliya2288 committed:
Домашнее задание спринт 19 v3
Попыталась исправить ошибки, не уверена в правильности.
Помогите разобраться, пожалуйста.
Добавила в reset в override fun prepare
остальные ошибки не понимаю из-за чего возникают

Nataliya2288 committed:
Домашнее задание спринт 19 v4

Nataliya2288 committed:
Домашнее задание спринт 19 v5

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/player/presentation/PlayerViewModel.kt:
```diff
@@ -1,104 +1,123 @@
+            setOnCompletionListener {
+                statePlayerLiveData.postValue(PlayerState.STATE_COMPLETED)
+            }
+        } ?: run {
```
run можно убрать

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/player/presentation/PlayerViewModel.kt:
```diff
@@ -1,104 +1,123 @@
+        return if (track == null) {
+            PlayerState.STATE_ERROR
+        } else {
+            if (!track.collectionName!!.isNotEmpty()) {
```
Можно заменить на
![image](https://github.com/user-attachments/assets/a862a01f-2f6c-4c00-862c-8b7b3a78d51f)

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -0,0 +1,302 @@
+            clearHistoryButton.setOnClickListener {
+                viewModel.clearHistoryList()
+                historyAdapter.notifyDataSetChanged()
+                historyWidget.visibility = View.GONE
```
![view](https://github.com/user-attachments/assets/191a9b9a-cb00-47ae-a96e-5c93c273d881)

Grigoriy-M commented on app/src/main/java/com/practicum/playlistmaker/search/ui/SearchFragment.kt:
```diff
@@ -0,0 +1,302 @@
+                viewModel.searchRequest(inputEditText.text.toString())
+            }
+
+            val textWatcher = object : TextWatcher {
```
![TextWatcher](https://github.com/user-attachments/assets/b8353cf4-f81e-443c-b0cd-dc4b6f910065)

Nataliya2288 committed:
Домашнее задание спринт 19 v6

Nataliya2288 committed:
Домашнее задание спринт 19 v7
